actors/builtin/multisig/multisig_actor.go

```diff
@@ -302,17 +320,20 @@ func (a Actor) RemoveSigner(rt vmr.Runtime, params *RemoveSignerParams) *adt.Emp
 
 	var st State
 	rt.State().Transaction(&st, func() {
-		if !isSigner(rt.ResolveAddress, &st, params.Signer) {
-			rt.Abortf(exitcode.ErrNotFound, "%s is not a signer", params.Signer)
+		isSigner, resolved, err := isSigner(params.Signer, st.Signers, rt.ResolveAddress, rt.Send)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to resolve to ID address")
+		if !isSigner {
+			rt.Abortf(exitcode.ErrForbidden, "%s is not a signer", params.Signer)
 		}
 
 		if len(st.Signers) == 1 {
 			rt.Abortf(exitcode.ErrForbidden, "cannot remove only signer")
 		}
 
 		newSigners := make([]addr.Address, 0, len(st.Signers))
+		// signers have already been resolved
 		for _, s := range st.Signers {
-			if !isAddressEqual(rt.ResolveAddress, s, params.Signer) {
+			if resolved != s {
 				newSigners = append(newSigners, s)
 			}
```

Comment:
Do we still need to iterate over st.Signers? The !isSigner check should be functionally identical.

Reply:
I am not sure what you mean here, This iteration is to get the Signers slice without the signer we need to remove.
Can you elaborate on what you mean here ?